ocs_ci/ocs/registry.py

```diff
@@ -68,10 +68,10 @@ def get_registry_pod_obj():
             break
         elif len(pod_obj) == 0:
             raise UnexpectedBehaviour("Image-registry pod not present")
-        elif iteration > 5:
+        elif iteration == 18:  # 18*10 = 180 secs.
```

Comment:
Why are we looping over `range(30)` if we stop at 18?

Reply:
No idea. Again, followed previous code. Should I change that?

Reply:
It appears that the logic here is simply waiting for the pod_obj list to be of length 1 within 180 seconds checking 10 times. We have a retry decorator that we can use to simplify this entire chunk of code.

Something like this perhaps:

```python
@retry(UnexpectedBehaviour, tries=18, delay=10, backoff=1)
def get_registry_pod_obj():
    pod_data = pod.get_pods_having_label(
        label='docker-registry=default',
        namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
    )
    pod_obj = [pod.Pod(**data) for data in pod_data]
    if len(pod_obj) != 1:
        raise UnexpectedBehaviour("Old registry pod not fully terminated")
    return pod_obj
```

The only difference here is that the original code will raise the `UnexpectedBehavior` if the `pod_obj` length is 0 immediately. In my code here, it will simply be caught by the retry decorator until all attempts are tried. If we want to maintain the functionality of raising the error immediately when the length is 0 we can simply raise a slightly different error that is not caught by the retry decorator.

I'm not saying we NEED to change this all right now, I just wanted to call out that this logic isn't all that clear at first glance and could be made much more simple with tools we already have in our utility module.

Reply:
All I wanted is to reduce the wait time. I think your suggestions are good, but should not hold this PR?

Reply:
Sure, that's fine. But you are changing the amount of times this loop is iterating and that count doesn't match the logic that stops the loop at 18. I think this is the right time to at least address that and figure out a way to make it consistent.

Reply:
I was aiming NOT to change the amount of time, but the amount of iterations and reduce the amount of time between iterations... Did I make a mistake here?

Reply:
You didn't since the logic was already weird to begin with. I'll just create an issue and handle it another time.